src/reference/asciidoc/rmi.adoc

```diff
@@ -75,3 +75,30 @@ The following code snippet shows the different configuration for an outbound rmi
     remote-channel="testChannel"
     host="localhost"/>
 ----
+
+=== Configuring with Java Configuration
+
+[source, java]
+----
+@Bean
+public RmiInboundGateway inbound() {
+    RmiInboundGateway gateway = new RmiInboundGateway();
+    gateway.setRequestChannel(requestChannel());
+    gateway.setRegistryHost("host");
+    gateway.setRegistryPort(port);
+    return gateway;
+}
+
+@Bean
+@ServiceActivator(inputChannel="inChannel")
+public RmiOutboundGateway outbound() {
+    RmiOutboundGateway gateway = new RmiOutboundGateway("rmi://host:port/"
+        + RmiInboundGateway.SERVICE_NAME_PREFIX + "remoteChannelName");
```

Comment:
How about adding a Lambda sample for the `RmiProxyFactoryBeanConfigurer`? :smile: 
For example for the same `ContextPropagatingRemoteInvocationFactory` injection...